### ISSUE TITLE
Add CpuCount and CpuPercent To Build

### ISSUE THREAD

**Issue by taylorb-microsoft** [moby/moby#38387]
_Redirecting an issue from the Windows container docs (https://github.com/MicrosoftDocs/Virtualization-Documentation/issues/928)_

Currently build accepts cpushares, cpusetcpus, cpuperiod, cpuquota but not the corresponding Windows args of CpuCount, CpuPercent of CreateContainer.  This issue to track adding those options to build.

Ref:
https://docs.docker.com/engine/api/v1.39/#operation/ImageBuild
https://docs.docker.com/engine/api/v1.39/#operation/ContainerCreate

**Comment by thaJeztah** [comment]
@tonistiigi

**Comment by timorkal** [comment]
Any news regarding this issue?

**Comment by yatima1460** [comment]
Yeah this is ridiculous, that on Windows with Hyper-V you can only build containers with max 2 cores

```
Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM mcr.microsoft.com/windows/servercore:ltsc2019
 ---> 4abfec8815bc
Step 2/2 : RUN powershell echo ${Env:NUMBER_OF_PROCESSORS}
 ---> Running in 7d13bf1287ca
2
Removing intermediate container 7d13bf1287ca
 ---> 72c9afae070c
Successfully built 72c9afae070c
```

**Comment by tim-oleksii** [comment]
256 CPU cores at my space are wasted - moby cannot scale.

**Comment by m0nzderr** [comment]
It looks like #43269 was closed withoud making it to the branch. Any news regarding this issue?

**Comment by slonopotamus** [comment]
Whoops, that was unintended. Reopened.

**Comment by helmesjo** [comment]
Any progress on this? Having 12 cores/ 24 threads at your disposal but `docker build` only using 2 is a quite serious bug. FWIW `docker run` uses all cores/threads.

**Comment by lacava** [comment]
any updates? kind of defeats the purpose of parallel stage builds

**Comment by avin3sh** [comment]
Sigh...

**Comment by ternium1** [comment]
Any updates on this one?

**Comment by slonopotamus** [comment]
Since 2023, things became even worse. Moby is currently moving to BuildKit on Windows and there isn't even a `docker buildx build --isolation process` option to bypass HyperV CPU/memory limits. Option to control the limits is also absent.

**Comment by ternium1** [comment]
and is there a plan to integrate those changes into buildkit?

**Comment by ffesgsgssvcs75** [comment]
Message received.

**Comment by widas5s9sfdv62lssok** [comment]
Thanks, I'll follow up.
